### PR TITLE
chore: drop support for node 10

### LINKED ISSUE #306: In the next major version, drop support for Node 10
Node 10 reached the end of its life on [April 30](https://github.com/nodejs/Release).

There's nothing pushing Helmet to drop support in the current version, but the next version should require a minimum of Node 12.

This task should be a matter of:

- Updating any CI scripts that need it
- Updating the `engines` field in `package.json`
- For all middlewares that don't need a major version bump, updating their package overrides to allow Node 10. This is because I don't want to release a major version of a sub-package just to rev the Node version.

If you do this, make sure to make the PR against the `v5.x` branch.

### PR DESCRIPTION
**Pull request by Fdawgs**
Closes #306

### CODE REVIEW

File: middlewares/content-security-policy/package-overrides.json [hunk 1]
```diff
@@ -7,5 +7,8 @@
   ],
   "description": "Content Security Policy middleware",
   "version": "3.4.0",
-  "keywords": ["express", "security", "content-security-policy", "csp", "xss"]
+  "keywords": ["express", "security", "content-security-policy", "csp", "xss"],
```
Comment (EvanHahn) [inline middlewares/content-security-policy/package-overrides.json:10]: Could you revert this file? Helmet v5 will include a breaking change in this package, so we'll want to require Node 12.

**Review by EvanHahn — CHANGES_REQUESTED** [review]
Looks great, thanks for doing this.

A few small things:

1. Could you also update [the `build-middleware-package` script](https://github.com/Fdawgs/helmet/blob/f5fdb7680b0144d4a0bbedf203f5ea67d184fed1/bin/build-middleware-package.js#L44-L46) to require Node 12 by default?
2. Could you add a changelog entry that says "Drop support for Node 10 and 11. Node 12+ is now required" under a "Removed" header?

File: middlewares/content-security-policy/package-overrides.json (hunk 1 above)
Comment (Fdawgs) [reply middlewares/content-security-policy/package-overrides.json]: Sure, reverted in https://github.com/helmetjs/helmet/pull/315/commits/f1114429efa5d0cad52b0b602e1f13381904f32a

**Comment by Fdawgs** [comment]
> 
>     2. Could you add a changelog entry that says "Drop support for Node 10 and 11. Node 12+ is now required" under a "Removed" header?

Just in the top-level changelog?

**Comment by EvanHahn** [comment]
I'd do this in the top-level changelog and the Content Security Policy changelog, because both of those packages will have major version bumps.

**Comment by EvanHahn** [comment]
Thanks! I've merged this into the `5.x` branch and this will be released with Helmet 5.

**Comment by EvanHahn** [comment]
I made a mistake when rebasing this so I cherry-picked it into f97c4516ecea22de62917d9e6fca29ff4ef7f552.

**Pull request merged**